ase/CMakeLists.txt

```diff
@@ -35,5 +38,23 @@ set(EXTRA_ASE_FILES
        ase_regress.sh)
 
 install(FILES ${EXTRA_ASE_FILES}
-        DESTINATION share/opae/ase
+        DESTINATION ${ASE_SHARE_DIR}
         COMPONENT aseextra)
+
+## Some ASE scripts are installed in bin
+set(PLATFORM_SCRIPTS
+        afu_sim_setup)
+
+foreach(SRC ${PLATFORM_SCRIPTS})
+  ## Substitute CMake variables of the form @<var>@ in the source file
+  ## and store the file in the build tree's bin directory.
+  configure_file(
+          scripts/${SRC} ${PROJECT_BINARY_DIR}/bin/${SRC}
+          @ONLY NEWLINE_STYLE UNIX)
+  ## This forces the configure_file to be built for all targets
+  add_custom_target(${SRC} ALL DEPENDS ${PROJECT_BINARY_DIR}/bin/${SRC})
```

Comment:
Is this necessary? Isn't `configure_file` always run as part of the cmake configure step?

Reply:
No, this is special. It performs variable substitution and puts the script in bin, where it is available to the testing environment even when not installed.